x-pack/test/api_integration/apis/transform/stop_transforms.ts

```diff
@@ -0,0 +1,182 @@
+        const reqBody: StopTransformsRequestSchema = [
+          { id: 'invalid_transform_id', state: 'started' },
+        ];
+        const { body } = await supertest
+          .post(`/api/transform/stop_transforms`)
+          .auth(
+            USER.TRANSFORM_POWERUSER,
+            transform.securityCommon.getPasswordForUser(USER.TRANSFORM_POWERUSER)
+          )
+          .set(COMMON_REQUEST_HEADERS)
+          .send(reqBody)
+          .expect(200);
+
+        expect(body.invalid_transform_id.success).to.eql(false);
+        expect(body.invalid_transform_id).to.have.property('error');
+      });
+    });
+
+    describe('bulk stop', function () {
+      const reqBody: StopTransformsRequestSchema = [
+        { id: 'bulk_stop_test_1', state: 'started' },
+        { id: 'bulk_stop_test_2', state: 'started' },
+      ];
+      const destinationIndices = reqBody.map((d) => generateDestIndex(d.id));
+
+      beforeEach(async () => {
+        await asyncForEach(reqBody, async ({ id }: { id: string }, idx: number) => {
+          await createAndRunTransform(id);
+        });
+      });
+
+      afterEach(async () => {
+        await transform.api.cleanTransformIndices();
+        await asyncForEach(destinationIndices, async (destinationIndex: string) => {
+          await transform.api.deleteIndices(destinationIndex);
+        });
+      });
+
+      it('should stop multiple transforms by transformIds', async () => {
+        const { body } = await supertest
+          .post(`/api/transform/stop_transforms`)
+          .auth(
+            USER.TRANSFORM_POWERUSER,
+            transform.securityCommon.getPasswordForUser(USER.TRANSFORM_POWERUSER)
+          )
+          .set(COMMON_REQUEST_HEADERS)
+          .send(reqBody)
+          .expect(200);
+
+        await asyncForEach(reqBody, async ({ id: transformId }: { id: string }, idx: number) => {
+          expect(body[transformId].success).to.eql(true);
+          await transform.api.waitForIndicesToExist(destinationIndices[idx]);
+        });
+      });
+
+      it('should stop multiple transforms by transformIds, even if one of the transformIds is invalid', async () => {
+        const invalidTransformId = 'invalid_transform_id';
+        const { body } = await supertest
+          .post(`/api/transform/stop_transforms`)
+          .auth(
+            USER.TRANSFORM_POWERUSER,
+            transform.securityCommon.getPasswordForUser(USER.TRANSFORM_POWERUSER)
+          )
+          .set(COMMON_REQUEST_HEADERS)
+          .send([
+            { id: reqBody[0].id, state: reqBody[0].state },
+            { id: invalidTransformId, state: 'stopped' },
+            { id: reqBody[1].id, state: reqBody[1].state },
+          ])
+          .expect(200);
+
+        await asyncForEach(reqBody, async ({ id: transformId }: { id: string }, idx: number) => {
+          expect(body[transformId].success).to.eql(true);
```

Comment:
We should also validate that the transform actually stopped.

Reply:
Fixed in efcf9b3.